client/txn.go

```diff
@@ -76,6 +76,7 @@ type Txn struct {
 	// is (still) unset.
 	haveTxnWrite bool
 	haveEndTxn   bool // True if there was an explicit EndTransaction
+	execing      bool // True if the txn is run in the exec() loop
```

Comment:
And how do you know whether to call `Commit` or `CommitNo{Abort,Rollback}`? Seems strange to expose that when the `Txn` itself knows which one you want.